### DISCUSSION TITLE
Does you client has a retry feature for the mirror or cp commands?

### DISCUSSION THREAD

**Discussion by AlexisEspinosaGayosso** [minio/minio#14158] [Q&A] [answered]
Does you client has a retry feature for the `mc mirror` or `mc cp` commands?

This feature helps when transfers fail for some external reason and clients are able to restart transfers again rather than aborting the command. This feature is present in `rclone sync`, and I'm wondering if a similar feature is already in place for `minio/mc`?

If not, could you briefly explain why? And what is the usual workaround alternative while still using mc? Also plans for the future?
If yes, could you briefly explain how to make it work? How to test it? And also link me to suitable documentation?

Many thanks

**Comment by harshavardhana** [comment, marked as answer]
Retry is automatically done minio-go library that is used by `mc` @AlexisEspinosaGayosso https://github.com/minio/minio-go/blob/master/retry.go

**Reply by AlexisEspinosaGayosso** [reply]
@harshavardhana , Thanks a lot for your quick response.
Please forgive my ignorance, so the retry already implemented on mc is really an equivalent to the `rclone sync --retries` option? This I mean in terms of "retrying".
Also, in terms of configurability, is `MaxRetry` a configurable variable while interacting/utilising mc? Or this is a fix parameter which I may need to modify directly on the source?
Many many thanks,
Alexis

**Reply by harshavardhana** [reply]
It is not configurable @AlexisEspinosaGayosso - it is automatically handled by `mc`

**Reply by MrScruffyish** [reply]
I added a few questions regarding this "retry" in this issue:  https://github.com/minio/mc/issues/4138#issuecomment-1262085690
